### PR TITLE
feat: check command to evaluate rules

### PR DESCRIPTION
**Pull request by jimschubert**
Implement the 'ossify check' command to validate a directory structure
against one or more convention rule sets.

Features:
- Check by convention name: ossify check Go
- Check by convention flag: ossify check -c 'Standard Distribution'
- Check by JSON file: ossify check -f my-convention.json
- Check all conventions: ossify check --all
- Specify target directory: ossify check Go -d /path/to/project

Rule evaluation:
- Required: must exist, fails if missing
- Prohibited: must not exist, fails if present
- Preferred: should exist, warns if missing
- Optional: may exist, always passes
- Pattern: glob matching for files (e.g., *.go)

Output:
- Visual indicators for pass/fail/warn/skip
- Summary with pass/fail/warn/skip counts
- Exit code 1 if any required rules fail

Implementation:
- Add Evaluate() method to Convention type
- Add RuleResult and CheckResult types
- Add evaluateRule() helper for individual rule checking
- Support Directory, File, and Pattern rule types

Tests:
- Comprehensive unit tests for Convention.Evaluate()
- Test all rule levels and types
- Test edge cases (file vs directory mismatches)
---

* TestCheckCmd_Help - tests help output
* TestLoadConventionFromFile - tests JSON loading
* TestLoadConventionFromFile_NotFound - tests missing file error
* TestFindConventionByName - tests convention loOKup
* TestCheckCmd_WithConventionFile - integration test
* TestCheckCmd_Integration - full integration test

### CODE REVIEW

File: internal/model/convention.go [hunk 1]
```diff
@@ -111,6 +113,182 @@ func (c *Convention) Print() error {
...
+		if ruleResult.Passed {
+			result.PassCount++
+		} else {
+			switch rule.Level {
+			case Required, Prohibited:
+				result.FailCount++
+			case Preferred:
+				result.WarnCount++
+			case Optional:
+				result.SkipCount++
```
[the remark this answers is not in the document]
Comment (jimschubert) [reply internal/model/convention.go:187]: it's dead code, recommendation would cause a linting issue.

File: cmd/check.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	}
+
+	return &convention, nil
+}
+
+func findConventionByName(conventions []model.Convention, name string) *model.Convention {
+	nameLower := strings.ToLower(name)
+	for _, c := range conventions {
+		if strings.ToLower(c.Name) == nameLower {
+			return &c
```
[the remark this answers is not in the document]
Comment (jimschubert) [reply cmd/check.go:204]: Go 1.25
